core/rawdb/bor_receipt.go

```diff
@@ -68,8 +68,18 @@ func ReadRawBorReceipt(db ethdb.Reader, hash common.Hash, number uint64) *types.
 	// Convert the receipts from their storage form to their internal representation
 	var storageReceipt types.ReceiptForStorage
 	if err := rlp.DecodeBytes(data, &storageReceipt); err != nil {
-		log.Error("Invalid receipt array RLP", "hash", hash, "err", err)
-		return nil
+		storageReceipts := []*types.ReceiptForStorage{}
+		if err := rlp.DecodeBytes(data, &storageReceipts); err != nil {
+			log.Error("Invalid bor receipt array RLP", "number", number, "hash", hash, "err", err)
+			return nil
+		}
+
+		if nReceipts := len(storageReceipts); nReceipts != 1 {
+			log.Error("Invalid bor receipt array RLP length", "number", number, "hash", hash, "nReceipts", nReceipts)
+			return nil
+		}
+
+		return (*types.Receipt)(storageReceipts[0])
```

Comment:
I could be wrong, but it seems that this change isn't relevant to the PR?

Reply:
Yes, you're right. I think Delweng (creator or original feature PR) took this from ReadReceipts function where we can have multiple receipts. W.r.t bor receipt, I think it's fine and we don't need this change. Thanks for pointing out.